src/mapgen.cpp

```diff
@@ -320,6 +322,54 @@ void Mapgen::updateHeightmap(v3s16 nmin, v3s16 nmax)
 	//printf("updateHeightmap: %dus\n", t.stop());
 }
 
+
+void Mapgen::calcSaoLimits()
+{
+	// Chunksize, in blocks
+	s16 csize_b = csize.X / MAP_BLOCKSIZE;
+	// Central chunk offset, in blocks
+	s16 ccoff_b = -csize_b / 2;
+	// Minp/maxp of central chunk, in nodes
+	s16 ccmin = ccoff_b * MAP_BLOCKSIZE;
+	s16 ccmax = ccmin + csize.X - 1;
+	// Fullminp/fullmaxp of central chunk, in nodes
+	s16 ccfmin = ccmin - MAP_BLOCKSIZE;
+	s16 ccfmax = ccmax + MAP_BLOCKSIZE;
+	// Effective mapgen limit, in blocks
+	// Use same calculation as ServerMap::blockpos_over_mapgen_limit(v3s16 p)
+	s16 mapgen_limit_b = rangelim(mapgen_limit,
+			0, MAX_MAP_GENERATION_LIMIT) / MAP_BLOCKSIZE;
+	// Effective mapgen limits, in nodes
+	s16 mapgen_limit_min = -mapgen_limit_b * MAP_BLOCKSIZE;
+	s16 mapgen_limit_max = (mapgen_limit_b + 1) * MAP_BLOCKSIZE - 1;
+	// Number of complete chunks from central chunk fullminp/fullmaxp
+	// to effective mapgen limits.
+	s16 numcmin = (ccfmin - mapgen_limit_min) / csize.X;
+	s16 numcmax = (mapgen_limit_max - ccfmax) / csize.X;
+	// Terrain edges, in nodes
+	s16 terrain_edge_min = ccmin - numcmin * csize.X;
+	s16 terrain_edge_max = ccmax + numcmax * csize.X;
+	// SAO position limits, in Irrlicht units
+	sao_limit_min = terrain_edge_min * BS - 3.0f;
+	sao_limit_max = terrain_edge_max * BS + 3.0f;
+
+	// For testing
+	printf("sao limit min: %f\n", sao_limit_min);
+	printf("sao limit max: %f\n", sao_limit_max);
+}
+
+
+const bool Mapgen::saoPosOverLimit(const v3f &p)
```

Comment:
Ok thanks, i appreciate your hints intended to help me learn how to code this correclty. but i barely understand that, it could take days for me to learn how to do this.
In the current situation where there is no time for learning what i need is someone to just write out the code needed. I don't expect you to do this if you don't want to,maybe someone else will help.